src/main/kotlin/org/opensearch/replication/action/stop/TransportUnfollowIndexReplicationAction.kt

```diff
@@ -0,0 +1,68 @@
+package org.opensearch.replication.action.stop
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.apache.logging.log4j.LogManager
+import org.opensearch.action.support.HandledTransportAction
+import org.opensearch.action.ActionRequest
+import org.opensearch.action.support.ActionFilters
+import org.opensearch.action.support.master.AcknowledgedResponse
+import org.opensearch.client.Client
+import org.opensearch.common.inject.Inject
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver
+import org.opensearch.cluster.service.ClusterService
+import org.opensearch.commons.replication.action.ReplicationActions.STOP_REPLICATION_ACTION_TYPE
+import org.opensearch.commons.replication.action.StopIndexReplicationRequest
+import org.opensearch.commons.replication.action.ReplicationActions.UNFOLLOW_REPLICATION_ACTION_NAME
+import org.opensearch.commons.utils.recreateObject
+import org.opensearch.core.action.ActionListener
+import org.opensearch.replication.metadata.ReplicationMetadataManager
+import org.opensearch.replication.util.coroutineContext
+import org.opensearch.replication.util.stackTraceToString
+import org.opensearch.replication.util.suspendExecute
+import org.opensearch.tasks.Task
+import org.opensearch.threadpool.ThreadPool
+import org.opensearch.transport.TransportService
+
+
+/**
+ * This action transforms the request from ActionRequest type to StopIndexReplicationRequest
+ * and performs the TransportStopIndexReplicationAction on it.
+ * While TransportStopIndexReplicationAction is used directly by the _stop replication REST API,
+ * this action is used for inter-plugin communication by ism plugin to unfollow i.e. stop replication.
+ */
+class TransportUnfollowIndexReplicationAction @Inject constructor (

```

Comment:
let's rename is to something like `TransportIsmStopIndexReplicationAction`. Basically remove `unfollow` and add ISM in action name to distinct it from other supported actions.

Reply:
Sure thanks. [Suggestions](https://github.com/opensearch-project/index-management/pull/1198#discussion_r1676208203) were given in the ISM PR to name this class as **InternalTransportStopIndexReplicationAction** and the action name as "indices:**internal**/plugins/replication/index/**stop**", so the name is differentiated to be something internally invoked from plugins. We hadn't added the name ism, so it remains generic to be used by something else too in future.
I hope these names shall be okay?

Reply:
`TransportInternalStopIndexReplicationAction` should be fine